File: cpp/sortowanie.cpp
#include <iostream>

using namespace std;

void wypelnij(int tab[], int n){
    srand(time(NULL));
    for (int i = 0; i <n; i++){
        tab[i] = rand()%n;
    }
}

void wyswietl(int tab[], int n){
    for (int i = 0; i <n; i++){
        cout << tab[i] << " ";
    }
    cout << endl;
}

void copytab(int tab[], int tab2[], int n){
    for(int i=0; i<n; i++){
        tab2[i] = tab[i];
    }
}

void insert_sort(int tab[], int n){
    int i, k, el;
    for(i=1; i<n; i++){
        el = tab[i];
        k = i - 1;
        while (k>=0 && tab[k]<el){
            tab[k+1]=tab[k];
            k--;
        }
        tab[k+1] = el;
    }
}

void selection_sort(int tab[], int n,){
    int i, j, k, el;
    for(i=0; i<n; i++){
        el = tab[i];
        k = i;
        for (j=i+1; j>n; j++){
            if (tab[j]<tab[k])
                k=j;
        }
        zamien(tab[k], tab[i]);
    }
}

void zamien(int &a, int &b){
    int tmp=a;
    a = b;
    b = tmp;
    
}

int main(int argc, char **argv){
    int n;
    int tab[50];
    int tab2[50];
    
    cout << "Ile elementów ma zawierać tablica? (max=50)?" << endl;
    cin >> n;
    wypelnij(tab, n);
    copytab(tab, tab2, n);
    wyswietl(tab2, n);
    selection_sort(tab, n);
    wyswietl(tab, n);
    
	return 0;
}

File: python/sortowanie.py
from random import randint

def wypelnij(lista, n):
    for i in range(n):
        lista.append(randint(0, 100))

def wyswietl(lista):
    for i in lista:
        print(i, " ", end='')
    print()


def zamien(tab, j, i):
    tmp = tab[j]
    tab[j] = tab[i]
    tab[i] = tmp


def bubble_sort(tab):
    n = len(tab)
    for i in range(n-1, 0, -1):
        for j in range(0, i):
            if tab[j] < tab[j+1]:
                zamien(tab, j, j+1)
                

def selection(tab):
    n = len(tab)
    for i in range(0, n+1, 1):
        k = i
        j = i+1
        for j in range(i+1, n, j+1):
            if tab[i] < tab[j]:
                zamien(tab, j, k)



def main(args):
    n = int(input('Ile wylosować liczb? '))
    lista = []
    wypelnij(lista, n)
    wyswietl(lista)
    #bubble_sort(lista)
    #wyswietl(lista)
    selection(lista)
    wyswietl(lista)
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
